cmd/root.go

```diff
@@ -1,61 +1,363 @@
 package cmd
 
 import (
-	"strings"
-
-	"os"
-
+	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"net/url"
+	"os"
+	"strings"
+	"time"
 
 	"github.com/Sirupsen/logrus"
+	"github.com/andygrunwald/go-jira"
 	"github.com/fsnotify/fsnotify"
+	"github.com/google/go-github/github"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	"golang.org/x/oauth2"
 )
 
 var (
-	log             *logrus.Logger
+	log             *logrus.Entry
 	defaultLogLevel = logrus.InfoLevel
 	rootCmdFile     string
 	rootCmdCfg      *viper.Viper
+
+	since               time.Time // The earliest GitHub issue updates we want to retrieve
+	ghIDFieldID         string    // The customfield ID of the GitHub ID field in JIRA
+	ghNumFieldID        string    // The customfield ID of the GitHub Number field in JIRA
+	ghLabelsFieldID     string    // The customfield ID of the GitHub Labels field in JIRA
+	isLastUpdateFieldID string    // The customfield ID of the Last Issue-Sync Update field in JIRA
+
+	project jira.Project
 )
 
 func Execute() {
 	if err := RootCmd.Execute(); err != nil {
-		fmt.Println(err)
+		log.Debugln(err)
 		os.Exit(1)
 	}
 }
 
+func GetGitHubClient(token string) (*github.Client, error) {
```

Comment:
I admit that I should have been documenting all of these as I went, but I did document them all in #11, if that helps.